src/internal/literals.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Returns true if input number type is a literal
+ */
+type IsNumberLiteral<T extends number> = [T] extends [number]
+  ? [number] extends [T]
+    ? false
+    : true
+  : false
+
+type IsBooleanLiteral<T extends boolean> = [T] extends [boolean]
+  ? [boolean] extends [T]
+    ? false
+    : true
+  : false
+
+/**
+ * Returns true if any elements in boolean array are the literal true (not false or boolean)
+ */
+type Any<Arr extends boolean[]> = Arr extends [
+  infer Head extends boolean,
+  ...infer Rest extends boolean[],
+]
+  ? IsBooleanLiteral<Head> extends true
+    ? Head extends true
+      ? true
+      : Any<Rest>
+    : Any<Rest>
+  : false
+
+/**
+ * Returns true if every element in boolean array is the literal true (not false or boolean)
+ */
+type All<Arr extends boolean[]> = IsBooleanLiteral<Arr[number]> extends true
```

Comment:
I used Any/All (python equivalent of some/every) by accident, but I left the name because I don't want people thinking it behaves like some/every.

Because we're doing an array check at the type level, it has to consider each of `true`, `false`, `boolean` -- which is slightly different than a runtime check of truthy/falsy values